### PR TITLE
Add slack failure notification to the deploy workflow

### PR DESCRIPTION
**Pull request by johnake**
### Context

A pipeline failure was picked up recently - there was no slack notification set up for this pipeline.
We need to add slack notifications for failures so that such issues are picked up quickly

### Changes proposed in this pull request

Add slack notification to build-and-deploy workflow

### Guidance to review

Notification successfully tested:

image

### Checklist

- [x] Attach to Trello card
- [x] Rebased main
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by johnake** [comment]
[FALTRN: Add slack failure notification to the deploy workflow](https://trello.com/c/JX4yhsEF/1636-faltrn-add-slack-failure-notification-to-the-deploy-workflow)

File: .github/workflows/build-and-deploy.yml [hunk 1]
```diff
@@ -78,19 +78,43 @@ jobs:
...
         id: smoke-test
         with:
           environment: ${{ matrix.environment }}
           azure_credentials: ${{ secrets.AZURE_CREDENTIALS }}
 
+      - name: Slack Notification
+        if: failure()
+        uses: rtCamp/action-slack-notify@master
+        env:
+          SLACK_COLOR: ${{ env.SLACK_FAILURE }}
```
Comment (malcolmbaig) [inline .github/workflows/build-and-deploy.yml:113]: SLACK_FAILURE doesn't appear to be set anywhere

**Review by malcolmbaig — COMMENTED** [review]
Nice, thanks for picking this up 👍🏼 

One comment from me about the SLACK_COLOR setting, but otherwise LG.

File: .github/workflows/build-and-deploy.yml (hunk 1 above)
Comment (johnake) [reply .github/workflows/build-and-deploy.yml:113]: Will remove - except if we think the colour  is required?

Comment (johnake) [reply .github/workflows/build-and-deploy.yml]: Fixed

**Review by saliceti — APPROVED** [review]

**Pull request merged**